pkg/profile/kustomize.go

```diff
@@ -0,0 +1,54 @@
+package profile
+
+import (
+	"reflect"
+	"time"
+
+	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KustomizeRequiresUpdate checks if the git kustomization resource requires updating
+func KustomizeRequiresUpdate(existingRes, desiredRes *kustomizev1.Kustomization) bool {
```

Comment:
Could this use `reflect.DeepEqual(desireRes.Spec, existingRes.Spec)`?

Reply:
good question. I leaned towards not doing `DeepEqual`  where there are fields in the struct that we don't specifically set. These fields might be getting defaulted or set by a webhook. Essentially we only check for updates to fields we specifically set.